lib/netext/httpext/request.go

```diff
@@ -116,6 +134,56 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 		Headers: preq.Req.Header,
 	}
 	if preq.Body != nil {
+		if len(preq.Compressions) > 0 {
+			var contentEncoding string
+			var prevBuf = preq.Req.Body
+			var buf *bytes.Buffer
+			for _, compressionType := range preq.Compressions {
+				if buf != nil {
+					prevBuf = ioutil.NopCloser(buf)
+				}
+				buf = new(bytes.Buffer)
+
+				if contentEncoding != "" {
+					contentEncoding += ", "
+				}
+				contentEncoding += compressionType.String()
+				var w io.WriteCloser
+				switch compressionType {
+				case CompressionTypeGzip:
+					w = gzip.NewWriter(buf)
+				case CompressionTypeDeflate:
+					// Have the compression level as an option somewhere ?
+					// the error is only possible if we provide value outside of the range [-2, 9]
+					w, _ = flate.NewWriter(buf, flate.DefaultCompression)
```

Comment:
I'm not sure `flate` is what we should use here. For the decompression, we use the `zlib` library: https://github.com/loadimpact/k6/blob/4894b66e892f4a37830fb7a9e62055f72940ec79/lib/netext/httpext/request.go#L247-L252
_(that code should probably also use the new constants and be encapsulated in a function that returns an `io.Reader`)_

I don't remember the reason, but I remember that there was a reason :smile: Maybe @robingustafsson can shed some light, I think this was the original PR that added support for deflate: https://github.com/loadimpact/k6/pull/419

Reply:
I don't have time to read through those RFCs at the moment, but according to [Wikipedia](https://en.wikipedia.org/wiki/HTTP_compression#Content-Encoding_tokens), `deflate` is 
> compression based on the deflate algorithm (described in RFC 1951), a combination of the LZ77 algorithm and Huffman coding, wrapped inside the zlib data format (RFC 1950);

Reply:
I reread it three times yesterday but apperantly you are right